src/modules/site-v2/templates/_scripts/gene-search.js

```diff
@@ -0,0 +1,76 @@
+/* Hide the gene search table, and show the loading icon if a search term has been entered.
+ *
+ * Arguments:
+ *   - gene: The gene search term
+ *   - table_id: The ID of the table element where results are stored.
+ *   - loading_id: The ID of the table loading icon. Displays while gene search is running.
+ */
+function prep_gene_search(gene, table_id, loading_id) {
+
+  // If gene is empty, hide all search elements
+  if (gene.trim().length == 0) {
+    $(table_id).fadeOut();
+    $(loading_id).fadeOut();
+  }
+
+  // If gene not empty, replace the table with the loading icon
+  else {
+    $(table_id).fadeOut().promise().done(() => { $(loading_id).fadeIn(); });
+  }
+}
+
+
+/* Query the db for genes and display in a results table.
+ *
+ * Arguments:
+ *   - gene: The gene to query for
+ *   - species: The species to query for
+ *   - table_id: The ID of the table element to store the results in. Clears & appends rows to the tbody(s) of this table.
+ *       NOTE: If used on a table with multiple tbody elements, will append to ALL bodies.
+ *   - loading_id: The ID of the table loading icon. Displays while gene search is running.
+ *   - callback: A function to run on each returned gene:
+ *       Arguments: element index & element (gene object)
+ *       Return: a list of object to be placed in table cells, or null if this gene should be skipped.
+ */
+function run_gene_search(gene, species, table_id, loading_id, callback) {
+
+  // Make sure all whitespace is trimmed
+  gene = gene.trim();
+
+  // Get the body of the table to fill out
+  const tbody = $(`${table_id} > tbody`);
+  console.log(tbody);
+  
+  // If no search term provided, hide both dropdown elements
+  if (gene.length == 0) {
+    $(table_id).fadeOut();
+    $(loading_id).fadeOut();
+  }
+  
+  // If search provided, query the database
+  else {
+    tbody.html("");
+    $.ajax({
+      url:         "{{ url_for('api_gene.api_search_genes', query='') }}" + gene + '?species=' + species.name,
+      method:      "GET",
+      contentType: 'application/xml',
+    }).done(function(results) {
```

Comment:
handle the promise exception/error (API issues, etc).

Reply:
Added a basic `console.error` handler in e0c14739b15ba3dea8061a3d82005aa5081cfbfb.  More rigorous error handling will be a bit tricky:

- Currently, we display errors to the user by flashing them in Flask and reloading the page -- this is a bit hard to replicate in JS.
- If we want to display to the user, we'll have to work out what an appropriate error message would be.

What would you recommend for handling errors from this call?